### PR TITLE
Update caching of tensor arguments for nvFuser's fusion creation

### PR DESCRIPTION
**Pull request by IvanYashchuk**
Previously nvFuser's fusion definition was cached based on concrete shape and strides of tensor inputs for simplicity and correctness. This PR changes Python's cache to check the number of dimensions, size-1 dimensions, and contiguity information based on given strides and shapes.

cc @kevinstephano @jjsjann123

### CODE REVIEW

**Review by kevinstephano — APPROVED** [review]
Nice addition! LGTM.

File: torch/csrc/jit/codegen/cuda/python_frontend/python_bindings.cpp [hunk 1]
```diff
@@ -39,6 +39,24 @@ void initNvFuserPythonBindings(PyObject* module) {
...
+         const std::vector<int64_t>& strides) {
+        py::tuple contiguity(sizes.size());
+        TORCH_CHECK(
+            sizes.size() == strides.size(),
+            "compute_contiguity: Sizes and strides must have the same number of dimensions");
+        if (sizes.size() == 0) {
+          return contiguity;
+        }
+        contiguity[sizes.size() - 1] = strides.back() == 1;
+        for (int64_t i = static_cast<int64_t>(sizes.size()) - 2; i >= 0; --i) {
```
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/python_frontend/python_bindings.cpp:54]: nitpick: `for (const auto i : c10::irange(sizes.size()-1)) {`

**Review by jjsjann123 — APPROVED** [review]
LGTM

**Comment by IvanYashchuk** [comment]
@mruberry could you please help merge this?

**Review by ngimel — APPROVED** [review]

**Comment by IvanYashchuk** [comment]
Dynamo failure is unrelated:
```py
  Error: alias_from_buffer_cpu_bool (__main__.TestAsArrayCPU) ... [2022-11-01 20:32:17,613] torch._dynamo.convert_frame: [ERROR] WON'T CONVERT <graph break in _check> test_tensor_creation_ops.py line 3676 
  due to: 
  Traceback (most recent call last):
    File "/opt/conda/lib/python3.7/site-packages/torch/_dynamo/variables/tensor.py", line 346, in create
      + f"{typestr(example_value)} {proxy.node.op} {proxy.node.target}"
  AssertionError: torch.* op returned non-Tensor int call_method data_ptr
  
  from user code:
     File "test_tensor_creation_ops.py", line 3680, in <graph break in _check>
      self.assertEqual(result.data_ptr(), original.data_ptr())
```

**Pull request closed without merging**